tests/components/nx584/test_binary_sensor.py

```diff
@@ -73,79 +79,81 @@ def test_setup_full_config(self, mock_nx, mock_watcher):
             "zone_types": {3: "motion"},
         }
         add_entities = mock.MagicMock()
-        assert nx584.setup_platform(self.hass, config, add_entities)
+        assert nx584.setup_platform(hass, config, add_entities)
         mock_nx.assert_has_calls(
             [
-                mock.call(self.fake_zones[0], "opening"),
-                mock.call(self.fake_zones[2], "motion"),
+                mock.call(fake_zones[0], "opening"),
+                mock.call(fake_zones[2], "motion"),
             ]
         )
         assert add_entities.called
         assert nx584_client.Client.call_count == 1
         assert nx584_client.Client.call_args == mock.call("http://foo:123")
         assert mock_watcher.called
 
-    def _test_assert_graceful_fail(self, config):
+    @staticmethod
+    async def _test_assert_graceful_fail(hass, config):
         """Test the failing."""
-        assert not setup_component(self.hass, "nx584", config)
-
-    def test_setup_bad_config(self):
+        assert not await async_setup_component(hass, "nx584", config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)